testing/select.test

```diff
@@ -57,4 +57,20 @@ do_execsql_test seekrowid {
 
 do_execsql_test select_parenthesized {
   select (price + 100) from products limit 1;
-} {179.0}
\ No newline at end of file
+} {179.0}
+
+do_execsql_test select_case_base_else {
+  select case when 0 then 0 when 1 then 1 else 2 end;
+} {1}
+
+do_execsql_test select_case_noelse_null {
+  select case when 0 then 0 end;
+} {}
+
+do_execsql_test select_base_case_else {
+  select case 1 when 0 then 0 when 1 then 1 else 2 end;
```

Comment:
i like that these tests are minimal and to the point. however i think they could be a bit more illustrative, for example:

```
select case 2 + 2 when 3 then 'first' when 4 then 'second' else 'third' end;
second
```

Reply:
on second thought I agree with what you said here: https://github.com/tursodatabase/limbo/pull/424#discussion_r1877298486

you're free to make a decision whether to make the tests more illustrative in some way or not, not a big concern

Reply:
I don't know how to make the null returning tests clearer because I fundamentally can't write anything there, but maybe the rest are more fun now.